napari/_qt/qt_resources/__init__.py

```diff
@@ -134,16 +134,19 @@ def import_resources(
     return respath, load
 
 
-@lru_cache(maxsize=4)
-def get_stylesheet(extra: Optional[List[str]] = None) -> str:
-    """Combine all qss files into single (cached) style string.
-
-    Note, this string may still have {{ template_variables }} that need to be
-    replaced using the :func:`napari.utils.theme.template` function.  (i.e. the
-    output of this function serves as the input of ``template()``)
+@lru_cache(maxsize=12)
+def get_stylesheet(
+    theme: str = None, extra: Optional[List[str]] = None
+) -> str:
```

Comment:
It looks like it changes current napari behavior. Currently `get_theme` and `register_theme` allow overwriting some theme parameters. I use it to overwrite the canvas background to have a black background with the `light` theme.

Reply:
The one thing this changes (due to caching) is that if someone were to modify a theme after setting that theme, those changes would not be reflected.  While that worked, I don't think we ever intended to "support" themes as mutable objects.  `register_theme` will allow you to create a new modified theme that will work here.  We understand you'd like to have the canvas background be settable apart from the theme, (#2103 and [mentioned above](https://github.com/napari/napari/pull/2263#issue-572951065)).  That's a good feature, very possible and unrelated to this.  We can enable that in another PR.

Reply:
I agree. But #2103 stuck with feedback that its needs to be discussed. If this PR will be merged and #2103 not then I lost functionality. 

And still, If both PR was merged then my current code may be broken. There may be missed one version to keep both versions working.

Reply:
> While that worked, I don't think we ever intended to "support" themes as mutable objects.

Yes, this was not an intended piece of functionality, but an unintended side effect.

I'd like to merge this PR now as it is improving the stylesheet getting. We can talk more about the canvas override in #2103. I think the best approach there might be a private attribute that you use until we have a public canvas object.

Reply:
The first discussion was in #2074 and @jni Told me that I could assume that theme is mutable
https://github.com/napari/napari/pull/2074#discussion_r551792658

Reply:
I think what @jni intended in that comment is to force registering of the new theme and not leverage the mutability of the dict, see also this https://github.com/napari/napari/pull/2013#discussion_r545142762. Personally I'd prefer us not to use mutable objects here, but I lost that discussion in #2013. We'll leave this open until @jni has a chance to respond too

Reply:
I still worry that there could be release with this PR and without possibility to set canvas color.

Reply:
no need to worry: https://github.com/napari/napari/pull/2270

Reply:
#2270 looks great, I just approved it. I think we now have a clear path forward to merge this PR followed by that one. I'll still wait in case anyone else has additional concerns, but I'm very happy with this approach